datamade/committee-oversight
# Fix Wagtail preview bug and import script

## Overview

Closes #93. This PR fixes a bug with draft previewing for local development. After some investigation, I discovered that the `Site` model should have a `pk` of 1 but had been mistakenly exported with a `pk` of 2. This came from a deeper problem, where the `load_cms_content` script was not cleaning up the Wagtail database before loading in initial data. This has led to unwanted primary key collisions and general messiness, and broken image links. Since we'll only use the script to move/recreate site data entirely, it's enough for this site to delete Wagtail models before importing `initial_cms_content.json`. This PR also fixes those broken images!

In the process of troubleshooting this, I updated Wagtail to the newest version (2.6.2).

## Testing Instructions

* Delete your volume: `docker-compose down --volumes`
* Start the site back up: `docker-compose up`
* Open a new window and load `hearings.dump`:  
`docker-compose exec postgres pg_restore -C -j4 --no-owner -U postgres -d hearings /app/hearings.dump`
* Run migrations: `docker-compose exec app python manage.py migrate`
* Load CMS content: `docker-compose run --rm app python manage.py load_cms_content`
* Load sample committee ratings: `docker-compose run --rm app python manage.py load_committeeratings`
* Log into the site at `http://localhost:8000/cms`
* Create a new page and make sure draft preview functionality works
* Check on the images and make sure they're displaying properly. You should see 3 images from Lugar's website: 
![Screen Shot 2019-10-08 at 4 26 37 PM](https://user-images.githubusercontent.com/1094243/66434576-75d3a580-e9e8-11e9-824e-216168655457.png)
